src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgeryLoggerExtensions.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Antiforgery.Internal
+{
+    internal static class DefaultAntiforgeryLoggerExtensions
+    {
+        private static readonly Action<ILogger, string, Exception> _validationFailed;
+        private static readonly Action<ILogger, Exception> _validating;
+        private static readonly Action<ILogger, string, Exception> _missingCookieToken;
+        private static readonly Action<ILogger, string, Exception> _missingFormToken;
+        private static readonly Action<ILogger, string, Exception> _missingHeaderToken;
+        private static readonly Action<ILogger, string, string, Exception> _missingRequestToken;
+
+        static DefaultAntiforgeryLoggerExtensions()
+        {
+            _validationFailed = LoggerMessage.Define<string>(
+                LogLevel.Warning,
+                1,
+                "Antiforgery validation failed with this message '{Message}'");
+            _validating = LoggerMessage.Define(
+                LogLevel.Information,
+                2,
+                "Antiforgery is attempting to validate a request.");
+            _missingCookieToken = LoggerMessage.Define<string>(
+                LogLevel.Warning,
+                3,
+                "The required antiforgery cookie '{CookieName}' is not present.");
+            _missingFormToken = LoggerMessage.Define<string>(
+                LogLevel.Warning,
+                4,
+                "The required antiforgery form field '{FormFieldName}' is not present.");
+            _missingHeaderToken = LoggerMessage.Define<string>(
+                LogLevel.Warning,
+                5,
+                "The required antiforgery header value '{HeaderName}' is not present.");
+            _missingRequestToken = LoggerMessage.Define<string, string>(
+                LogLevel.Warning,
+                6,
+                "The required antiforgery request token was not provided in either form field '{FormFieldName}'"
+                    + "or header value '{HeaderName}'.");
+        }
+
+        public static void ValidationFailed(this ILogger logger, AntiforgeryValidationException exception)
+        {
+            _validationFailed(logger, exception.Message, exception);
```

Comment:
If you're recalling a discussion between lou and I, no that was about something related but not this.